### PR TITLE
Feat/Metamask function component

### LINKED ISSUE #672: Metamask does not auto detect Boba tokens, thus will need a button on the gateway for users to easily add it to metamask to track

### PR DESCRIPTION
**Pull request by alvaro-ricotta**
:clipboard: Add associated issues, tickets, docs URL here.

## Overview

1) Adding metamask component
2) Refactory getCoinImage component

### CODE REVIEW

File: packages/boba/gateway/src/actions/networkAction.js [hunk 1]
```diff
@@ -18,6 +18,11 @@ import networkService from 'services/networkService'
 import transctionService from 'services/transaction.service'
 import { createAction } from './createAction'
 
+
+export function getWalletType(){
+  return networkService.walletService.walletType;
+}
+
```
Comment (sk-enya) [inline packages/boba/gateway/src/actions/networkAction.js]: Just to note is action function is something which has been dispatch so can you please move this function to service itself?

File: packages/boba/gateway/src/selectors/queueSelector.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/*
```
Comment (sk-enya) [inline packages/boba/gateway/src/selectors/queueSelector.js]: This is deleted file in develop branch so please remove!

File: packages/boba/gateway/src/reducers/feeReducer.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (sk-enya) [inline packages/boba/gateway/src/reducers/feeReducer.js]: This file is deleted in develop please remov.

File: packages/boba/gateway/src/hooks/useMetamask.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (sk-enya) [inline packages/boba/gateway/src/hooks/useMetamask.js]: Please remove.

File: packages/boba/gateway/src/components/global/addToMetamask/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { getWalletType } from 'actions/networkAction';
+import { getCoinImage } from 'util/coinImage';
+import MetamaskLogo from 'images/metamask.svg';
+
+export const AddToMetamask = ({ token }) => {
+  const { ethereum } = window;
+  const { address, symbol, decimals, chain } = token || {};
+  const walletType = getWalletType();
```
Comment (sk-enya) [inline packages/boba/gateway/src/components/global/addToMetamask/index.js]:
```suggestion
  const walletType = networkService.walletService.walletType
```

File: packages/boba/gateway/src/components/global/addToMetamask/index.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+              image: logoURI,
+              chainId: chain,
+            },
+          },
+        })
+        .catch((error) => {
+          console.error(error);
+        });
+    }
+  };
```
Comment (sk-enya) [inline packages/boba/gateway/src/components/global/addToMetamask/index.js]:
```suggestion
  const handleAddToMetamask =  async (token) => {
    await wallet.service.addTokenToMetaMask(token) /// let's add this function in service.
  }
```

**Review by sk-enya — APPROVED** [review]

**Review by souradeep-das — APPROVED** [review]

**Pull request merged**